lib/pure/terminal.nim

```diff
@@ -16,6 +16,33 @@
 ## Similarly, if you hide the cursor, make sure to unhide it with
 ## `showCursor` before quitting.
 
+##[
+## Playing with colorful and styled text
+]##
+
+## Procs like `styledWriteLine`, `styledEcho` etc. have a temporary effect on
+## the terminal. Style parameters only affect the text parameter right after them.
```

Comment:
this is either incorrect or misleading, for similar reason (no state stack) as https://github.com/nim-lang/Nim/pull/18296/files#r654584360:
```nim
when defined case2:
  import std/terminal
  proc main()=
    echo "ok1"
    stdout.setStyle({styleStrikethrough})
    echo "ok2"
    stdout.styledWriteLine(fgRed, "red text")
    echo "ok3"
    stdout.resetAttributes()
    echo "ok4"
  main()
```

shows (on osx):

![image](https://user-images.githubusercontent.com/2194784/122600480-af194100-d024-11eb-880b-446f59a7a3be.png)

=> the style of ok3 does not match the style of ok2

what actually happens is that `styledWriteLine` resets the default style after the function call returns, instead of the style that existed before calling this, so the effect is not localized

Reply:
I see, thanks. Now I clarify it a bit.